test/extended/testdata/storage/csi-secret-store/e2e-provider-installer.yaml

```diff
@@ -0,0 +1,52 @@
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    app: csi-secrets-store-e2e-provider
+  name: csi-secrets-store-e2e-provider
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: csi-secrets-store-e2e-provider
+  template:
+    metadata:
+      labels:
+        app: csi-secrets-store-e2e-provider
+    spec:
+      serviceAccountName: default
+      containers:
+        - name: e2e-provider
+          image: quay.io/jdobson/secrets-store-csi-driver-e2eprovider
```

Comment:
There is a process how to start using a new image in tests: https://github.com/openshift/origin/blob/master/test/extended/util/image/README.md
It would be great if we did not need to go through it.

Reply:
I assume a generic centos image + a shell script is not possible because of grpc. Then we need a new image + some openshift/ github repo to build it from it. Can we add it to the operator repo and build it in CI?

Reply:
I was originally trying to find a way to pass an image (built from CI) to openshift-tests, but this is not the right way to do it. IMO, the e2eprovider needs to be built from the driver repo first and then deployed by a script in step-registry before openshift-tests runs.

So I have a couple of PR's now to build e2eprovider from the driver repo:
https://github.com/openshift/secrets-store-csi-driver/pull/5
https://github.com/openshift/release/pull/41693
Once both of those are merged, I think I can rework this PR to deploy e2eprovider as part of the e2e workflow:
https://github.com/openshift/release/pull/41655
And if all of that actually works, then I'll update this PR so that openshift-tests does not try to deploy this anymore.

Reply:
> I assume a generic centos image + a shell script is not possible because of grpc.

This might be possible with a python script, but I haven't tried it. I can try that approach if the e2eprovider image turns out not to work for some reason.